lib/streamlit/cli.py

```diff
@@ -186,23 +198,13 @@ def main_run(file_or_url, args=None, **kwargs):
     _apply_config_options_from_cli(kwargs)
 
     if url(file_or_url):
-        import tempfile
-        import requests
-
-        with tempfile.NamedTemporaryFile() as fp:
-            try:
-                resp = requests.get(file_or_url)
-                resp.raise_for_status()
-                fp.write(resp.content)
-                # flush since we are reading the file within the with block
-                fp.flush()
-            except requests.exceptions.RequestException as e:
-                raise click.BadParameter(
-                    ("Unable to fetch {}.\n{}".format(file_or_url, e))
-                )
-            # this is called within the with block to make sure the temp file
-            # is not deleted
-            _main_run(fp.name, args)
+        from streamlit.temporary_directory import TemporaryDirectory
+        with TemporaryDirectory() as temp_dir:
+            from urllib.parse import urlparse
+            path = urlparse(file_or_url).path
+            script_path = os.path.join(temp_dir, path.rsplit('/', 1)[-1])
```

Comment:
Not sure there is a better way of doing this. This way seems "StackOverflow" accepted: https://stackoverflow.com/questions/7253803/how-to-get-everything-after-last-slash-in-a-url

Reply:
You should do `path.strip('/')` first to remove the final slash, if any.